### PR TITLE
Clean up Trajectory.append

### PR DESCRIPTION
**Pull request by eb8680**
This refactoring PR moves `Trajectory.append` into `chirho.dynamical.handlers.trace` and makes it slightly more generic.

### CODE REVIEW

File: chirho/dynamical/handlers/trace.py [hunk 1]
```diff
@@ -9,7 +10,43 @@
 T = TypeVar("T")
 
 
+@functools.singledispatch
+def append(fst, rest: T) -> T:
```
Comment (SamWitty) [inline chirho/dynamical/handlers/trace.py:14]: Should this be `append(fst: T, rest: T) -> T:`?

**Review by SamWitty — CHANGES_REQUESTED** [review]
One tiny comment. Otherwise looks great.

File: chirho/dynamical/handlers/trace.py (hunk 1 above)
Comment (eb8680) [reply chirho/dynamical/handlers/trace.py:14]: No, that doesn't work with `singledispatch`, which expects registered types to be subtypes of the original type on the first argument.  It's an annoying aspect of the way `singledispatch` interacts with `mypy`.

**Review by SamWitty — APPROVED** [review]
lgtm

**Pull request merged**